### DISCUSSION TITLE
Which TOTP apps can we use?

### DISCUSSION THREAD

**Discussion by DGrothe-PhD** [community/community#63742] [Other Feature Feedback, Questions, & Ideas] [answered]
### 2FA and apps

Question

### Body

Well, it’s gonna be mandatory to use 2FA on GitHub quite soon… 
I would like to use a TOTP app.
Does anyone know a suitable TOTP app?
Please name and give a link to both app stores (iOS and Android). Thanks in advance!

**Comment by 0xChintan** [comment]
Here are two popular TOTP (Time-Based One-Time Password) apps that you can use for 2FA on GitHub, available on both iOS and Android:

1. Google Authenticator
2. Authy


You can choose either of these apps, depending on your preference. They both work well for generating the time-based codes required for 2FA on GitHub.

**Reply by oniscoid** [reply]
authy.com just keeps giving me a 522 error -- doesn't inspire confidence

**Comment by TianMeds** [comment]
Another suggestion is the 

1. Microsoft Authenticator
2. FreeOTP Authenticator:

These apps can be used to set up 2FA for your GitHub account and also include TOTP capabilities. Pick the option that most closely matches your interests and needs. Follow best practices to keep your TOTP codes secure and frequently backup your 2FA codes, just like you would with any security-related software.

**Reply by lg2de** [reply]
I would like use FreeOTP but run into [freeotp/freeotp-android/issues/334].
Any suggestion?

**Comment by anztim** [comment]
Personal suggestion:

1. Microsoft Authenticator (especially when you already use it as your Microsoft account's 2FA)
2. Google Authenticator (my first OTP app)
3. keepass (It's a password manager. You can improve its OTP experience by adding extensions or using client like KeePassXC)

**Reply by mohameed88178h-hub** [reply]
Uytrewqsaf

**Comment by frakier** [comment]
Like everyone walks around with a phone attached to their rear. All the "apps" do not help when they all require a smartphone.

**Reply by queenofcorgis** [reply, marked as answer]
Hi there @frakier - I'm also someone who doesn't keep a phone on me most of the time so I get it. Luckily, there are plenty of options for desktop TOTP apps. 1Password is one example, but a search for `desktop TOTP apps` will reveal even more options.

**Reply by frakier** [reply]
will try adding desktop to the search, thanks. I found one for the desktop but it was windows not linux. But I was not using desktop in the search just a pure accidental find.

**Reply by queenofcorgis** [reply]
@frakier you're welcome - if you end up liking it - feel free to recommend to others here as well 😄

**Comment by Fish-Git** [comment]
> Like everyone walks around with a phone attached to their rear. All the "apps" do not help when they all require a smartphone.

_Exactly!_ Kids these days think everyone has a smart device (mobile/cell phone).

I don't have one and don't want one. I have an ordinary land-line telephone, and that's good enough for me.

I cannot do _(nor would I want to do)_ SMS (i.e. text messaging). I use email and that's good enough for me.

Because of this I am now suddenly being punished by no longer allowing me to manage nor contribute to an Open Source project I have been actively developing code for, for over the past 20+ years.

2FA should _NOT_ be required. It should be _optional_.

_**Fuck you, GitHub.**_

**Comment by Megidd** [comment]
~~For desktop operating systems like Linux, macOS, and Windows: https://github.com/arcanericky/totp~~

~~I tried it and looks amazing ✔️~~

<kbd>Update</kbd> Eventually, now I'm using a Firefox extension that is working just great. Firefox authenticator extension:

![2024-02-25 21_48_18-](https://github.com/community/community/assets/17475482/eb90b729-956c-4459-98c6-87fb3b608c0c)

![2024-02-25 21_48_52-](https://github.com/community/community/assets/17475482/18c40a48-b464-4cd3-a826-814391b14f04)

**Comment by fasakinhenry** [comment]
The thing is simple and easy to use, especially without needing a mobile phone. You can do this using a Chrome extension for people using Chrome on their PCs. 
- Go to https://authenticator.cc and add the authenticator extension to Chrome and grant all the permissions.
- Also grant all the required permissions
- Click the scan icon in the extension UI and scan the QR code given by GitHub or add the code manually.
- Enter the code it generates and voila you can add the code
- Lastly, save your recovery codes in a safe haven

**Comment by hellium6** [comment]
A no app solution would be to use: <https://a99us.github.io/totp-js/>

A "Pure JavaScript TOTP Token Generator".

It works with GitHub when many other TOTP apps and websites I tried failed. The page recommends saving it offline.

**Comment by tuves** [comment]
Thank you for your post @DGrothe-PhD! Upon reading, this appears to be a discussion with multiple opinions and no "right answer" as it's dependent on the user's choice. I've gone ahead and re-labeled this discussion so it can get the correct visibility.
